index.d.ts

```diff
@@ -10,6 +10,7 @@ declare class Agent implements Taggable, StartSpanFn {
   // Configuration
   start (options?: AgentConfigOptions): Agent;
   isStarted (): boolean;
+  getServiceName (): string;
   setFramework (options: {
```

Comment:
nit: Can also return _undefined_ if the configuration value isn't set.  _makes noises about the herculean task typescript has set for itself in typing the world's most dynamic mainstream programming language_

Reply:
Thanks, I'll update. You have to try hard, but yes it can be undefined.  The Agent actually loads config once before `.start()` and then again on `.start()`.  That first load was added a long while back just to be able to get create an internal logger to pass around.

That first "load config before .start()" will load `serviceName` from the following:
- the `ELASTIC_APM_SERVICE_NAME` env var, if it exists
- looking in $CWD for a "elastic-apm-node.js" file to `require()`
- looking up $CWD for a "package.json"

If none of those, then `serviceName` is undefined.

Unrelated to this change, but a fun breakage example:

```
% pwd
/

% ls ~/el/apm-agent-nodejs3/tmp
bar.js              elastic-apm-node.js

% cat ~/el/apm-agent-nodejs3/tmp/bar.js
var apm = require('../')
console.warn('XXX before', apm.getServiceName())
apm.start({
  serviceName: 'foo'
})
console.warn('XXX after', apm.getServiceName())

% node ~/el/apm-agent-nodejs3/tmp/bar.js
XXX before undefined
XXX after foo

% ELASTIC_APM_SERVICE_NAME="fromenv" node ~/el/apm-agent-nodejs3/tmp/bar.js
XXX before fromenv
XXX after fromenv

% ELASTIC_APM_SERVICE_NAME="" node ~/el/apm-agent-nodejs3/tmp/bar.js
XXX before undefined
Elastic APM isn't correctly configured: Missing serviceName
XXX after undefined
```

I found it surprising that a defined but empty-string envvar overrides the `serviceName` provided in the options to `.start()`.